Feature/notification-call-code

free2fqn opened the pull request:
Добавил функционал для верификации номера. На контроллер посылается запрос с текстом номера в формате "7*********" (как я понял можно даже и не 7, просто для других стран чуть дороже). Контроллер возвращает код безопасности для верификации, а ответ сервиса sms.ru печатает в консоль (удобно, что бы отслеживать баланс и стоимость).

free2fqn committed:
branch renamed to feature/notification-call-code. Added CodeCallController. Added field sms.api.id at application.yml

free2fqn committed:
created CodeCallResponseDTO for responses of sms.ru.
Created CodeCallService and it's CodeServiceImpl.
Created NotificationAppConfig for creating Beans.
Added field sms.url in application.yml.
Added endpoint "/callcode" in CodeCallController.

free2fqn committed:
created CodeCallSecurityCodeDTO. Reworked CodeCallController: '/codecall' now on GET and returns CodeCallSecurityCodeDTO. CodeCallService(Impl).sendGet now returns String. CodeCallResponseDTO clean up

nekromant322 commented on platform_notification/src/main/java/com/override/controllers/CodeCallController.java:
```diff
@@ -0,0 +1,26 @@
+    private final CodeCallService codeCallService;
+
+    @Autowired
+    CodeCallController(CodeCallService codeCallService) {
```
public?

free2fqn replied:
@nekromant322 Зачем? 0о

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/service/CodeCallService.java:
```diff
@@ -0,0 +1,7 @@
+
+public interface CodeCallService {
+
+    String sendGet(String clientPhoneNumber);
```
verifyNumber?

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/controllers/CodeCallController.java:
```diff
@@ -0,0 +1,26 @@
+        this.codeCallService = codeCallService;
+    }
+
+    @GetMapping("/codecall")
```
camelCase?

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/service/CodeCallServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+
+
+    @Autowired
+    CodeCallServiceImpl(RestTemplate restTemplate, ObjectMapper objectMapper) {
```
public?

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/service/CodeCallServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+            CodeCallResponseDTO codeCallResponseDTO = objectMapper.readValue(response.getBody(), CodeCallResponseDTO.class);
+            securityCode = codeCallResponseDTO.getCode();
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```
так не пойдет, пусть отлетает 500 от сервера

nekromant322 commented on platform_notification/src/main/java/com/override/service/CodeCallServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    public String sendGet(String clientPhoneNumber) {
+        String securityCode = "-1";
```
эм?

free2fqn replied:
идея была в том, что бы -1 была заглушкой и мы сразу понимали, что секьюрити код нормально не сгенерился

nekromant322 replied:
идея хуевая

free2fqn replied:
fixed

nekromant322 commented on platform_notification/src/main/java/com/override/service/CodeCallServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+        ResponseEntity<String> response = restTemplate.getForEntity(
+                url + "?phone=" + clientPhoneNumber + "&api_id=" + apiID,
+                String.class);
+        System.out.println(response.getBody());
```
убрать

free2fqn replied:
fixed

nekromant322 commented:
заменить  restTemplate на фейн

free2fqn committed:
clean up for 6 pull request

free2fqn committed:
Created BalanceResponseDTO.
Added spring-cloud-starter-feign and FeignClient support.
Reworked CodeCallService for FeignClient (created SmsRuFeign) instead of RestTemplate.
Added getBalance() to CodeCallService
In CodeCallController added '/balance' endpoint.
Added GetBalanceException and VerifyCallException.
Cutted sms.url field in application.yml.

free2fqn committed:
clean up

free2fqn commented:
Заменил на Feign

free2fqn committed:
clean up

nekromant322 commented on platform_notification/src/main/java/com/override/controller/CodeCallController.java:
```diff
@@ -0,0 +1,30 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
тут бы добавить мапинг какой-то, а то из корня торчит /codeCall и /balance  
я бы добавил /calls или что-то вроде того

nekromant322 commented on platform_notification/src/main/java/com/override/feign/SmsRuFeign.java:
```diff
@@ -0,0 +1,14 @@
+@FeignClient(value="smsRuFeignClient", url="${sms.url}")
+public interface SmsRuFeign {
+    @GetMapping("/code/call?phone={phone}&api_id={apiID}")
+    String verifyPhone(@PathVariable String phone, @PathVariable String apiID);
```
почему сразу не получать СodeCallResponseDTO?  
`СodeCallResponseDTO verifyPhone(@PathVariable String phone, @PathVariable String apiID);`

nekromant322 replied:
`@Pathvariable` не совсем корретно, можно юзать `@RequestParam("phone")`, тогда параметры сами к итоговой строке гет запроса добавится, а `@GetMapping("/code/call")`

nekromant322 commented on platform_notification/src/main/java/com/override/feign/SmsRuFeign.java:
```diff
@@ -0,0 +1,14 @@
+    String verifyPhone(@PathVariable String phone, @PathVariable String apiID);
+
+    @GetMapping("/my/balance?api_id={apiID}&json=1")
+    String getBalance(@PathVariable String apiID);
```
аналогично

free2fqn replied:
Здесь я считаю лучше оставить @PathVariable, иначе придется добавлять второй аргумент для флага json=1. Методом getBalance станет не так удобно пользоваться

nekromant322 replied:
вполне удобно, в сервисе будет константа с говорящим названием, которую при желании можно будет поменять

free2fqn committed:
removed ObjectMapper bean-field from CodeCallServiceImpl. Beacuse now methods in SmsRuFeign returns it's DTOs instead of Strings. Reworked endpoint-mapping in CodeCallController
